spot_wrapper/spot_arm.py

```diff
@@ -0,0 +1,561 @@
+                self._logger.info("Command gripper open angle sent")
+                time.sleep(2.0)
+
+        except Exception as e:
+            return False, "Exception occured while gripper was moving"
+
+        return True, "Opened gripper successfully"
+
+    def hand_pose(self, data) -> typing.Tuple[bool, str]:
+        try:
+            success, msg = self.ensure_arm_power_and_stand()
+            if not success:
+                self._logger.info(msg)
+                return False, msg
+            else:
+                pose_point = data.pose_point
+                # Move the arm to a spot in front of the robot given a pose for the gripper.
+                # Build a position to move the arm to (in meters, relative to the body frame origin.)
+                position = geometry_pb2.Vec3(
+                    x=pose_point.position.x,
+                    y=pose_point.position.y,
+                    z=pose_point.position.z,
+                )
+
+                # # Rotation as a quaternion.
+                rotation = geometry_pb2.Quaternion(
+                    w=pose_point.orientation.w,
+                    x=pose_point.orientation.x,
+                    y=pose_point.orientation.y,
+                    z=pose_point.orientation.z,
+                )
+
+                seconds = data.duration
+                duration = seconds_to_duration(seconds)
+
+                # Build the SE(3) pose of the desired hand position in the moving body frame.
+                hand_pose = geometry_pb2.SE3Pose(position=position, rotation=rotation)
+                hand_pose_traj_point = trajectory_pb2.SE3TrajectoryPoint(
+                    pose=hand_pose, time_since_reference=duration
+                )
+                hand_trajectory = trajectory_pb2.SE3Trajectory(
+                    points=[hand_pose_traj_point]
+                )
+
+                arm_cartesian_command = arm_command_pb2.ArmCartesianCommand.Request(
+                    root_frame_name=data.frame,
+                    pose_trajectory_in_task=hand_trajectory,
+                    force_remain_near_current_joint_configuration=True,
+                )
+                arm_command = arm_command_pb2.ArmCommand.Request(
+                    arm_cartesian_command=arm_cartesian_command
+                )
+                synchronized_command = (
+                    synchronized_command_pb2.SynchronizedCommand.Request(
+                        arm_command=arm_command
+                    )
+                )
+
+                robot_command = robot_command_pb2.RobotCommand(
+                    synchronized_command=synchronized_command
+                )
+
+                command = RobotCommandBuilder.build_synchro_command(robot_command)
+
+                # Send the request
+                self._robot_command_client.robot_command(robot_command)
+                self._logger.info("Moving arm to position.")
+
+                time.sleep(2.0)
```

Comment:
Why is there a sleep in this function?

Reply:
I can't be sure and I've not used the arm at all, but perhaps it's necessary to make sure the arm is actually in position before the command returns? It's not a good way to do this though. Perhaps there is a way to check the arm command feedback. The status message at https://dev.bostondynamics.com/protos/bosdyn/api/proto_reference#robotcommandfeedbackstatus-status might be a mechanism? I did a similar thing with trajectory commands.

Reply:
There  is a [`block_until_arm_arrives`](https://github.com/boston-dynamics/spot-sdk/blob/d1d743bbe419fddecea6c58865a681f19080395a/python/bosdyn-client/src/bosdyn/client/robot_command.py#L1731) function in the spot-sdk, which just polls the status messages. I think it would be better to use this function or write our own version.

Reply:
Good find, we should use the SDK functionality rather than implementing our own.